pandas/tests/test_algos.py

```diff
@@ -788,7 +787,8 @@ def test_datetime_likes(self):
             tm.assert_numpy_array_equal(res_false, exp_false)
 
             # index
-            for idx in [pd.Index(case), pd.Index(case, dtype='category')]:
+            for idx in [pd.Index(case), pd.Index(case, dtype='category'),
+                        pd.Index(case, dtype=object)]:
```

Comment:
Unrelated to the fix itself, but adding more tests to ``duplicated`` (this test revealed the bug).